Documentation/general/cli-ux-guidelines.md

```diff
@@ -0,0 +1,155 @@
+.NET Core Command-Line Tools UX Guidelines
+-------------------------------------------
+
+This document will cover the CLI UX guidelines that anyone adding or suggesting
+a new command should take into consideration. It does not deal with
+implementation details. The main goal of the document is to provide guidelines
+for new authors that wish to make pull requests and add new functionalities and
+commands to the .NET Core command line tools (CLI).
+
+## Naming the commands
+
+In the .NET Core CLI, commands should be **verbs**. This rule was adopted
+because most of the commands do *something*.
+
+Sub-commands are supported in the CLI, and they are usually nouns. A good
+example of this is the “dotnet add reference” command. If there is a need to add
+a subcommand, that subcommand should usually specialize what the parent command
+does.
+
+## CRUD commands
+If a new CRUD command is to be added, its naming should be determined with the
+following logic:
+
+-   Does the command work on the data in the project (either properties or
+    items)? If yes, then it should be added as a noun to the “dotnet
+    add/list/remove/update”, e.g. “dotnet add foo”.
+
+-   Does the command work on the solution (SLN) file? If yes, then it should be
+    added as a verb to the “dotnet sln” command.
+
+-   Does the command work on a completely new artifact (e.g. a new metadata file
+    that is added in the future)? If yes, it should be created as a top level
+    noun with all of the underlying operations defined as sub-commands.
+
+-   If the command adds a new artifact to the project, it should become an item
+    template that is dropped with “dotnet new” command.
+
+This setup was adopted to provide a clear namespacing that the users can
+remember and that fits well with usage patterns observed in other CLI toolsets a
+well as Visual Studio.
+
+If none of the above applies, the proposal should clearly outline why none of

```

Comment:
This is the first time we've seen "proposal". Oh, I'm making a proposal!?

How about:

> If none of the guidelines above apply, then a new guideline should be proposed and defined. This includes establishing why this case is unique.

This case seems to come in the middle of the list. Shouldn't it be last? Or should the following guidelines be in a different section?